### DISCUSSION TITLE
Change gradient surfacr parameters

### DISCUSSION THREAD

**Discussion by GINNMAX** [MREYE-LUMC/ZOSPy#48] [Q&A] [answered]
when i insert a gradient surface(gradient 3)，i tried to set Nr and Nz in python,but it cannot success in zemax optics.
i want to know the Nr and Nz's variable name in python api

**Comment by crnh** [comment, marked as answer]
Hi,

Due to missing example code, your question is not clear. Nevertheless, this looks like a generic ZOS-API question, so please ask this question on the [Zemax community forum](https://community.zemax.com/). It is not our intention to become a second OpticStudio forum and asking this question there is beneficial for the broader OpticStudio community.

Based on the information you provided, this topic on the Zemax community forum probably answers your question: https://community.zemax.com/zos-api-12/call-a-parameter-by-its-name-in-the-zos-api-1877

If this does not answer your question, please ask any further questions on the Zemax community forum.
